plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeSystemTables.java

```diff
@@ -99,4 +108,353 @@ public void testPropertiesTable()
+            assertUpdate("UPDATE " + tableName + " SET _bigint = 50 WHERE _bigint =  BIGINT '5'", 1);
+            assertUpdate("DELETE FROM " + tableName + " WHERE _date =  DATE '2019-09-08'", 1);
+            assertQuerySucceeds("ALTER TABLE " + tableName + " EXECUTE OPTIMIZE");
+            assertQuery("SELECT count(*) FROM " + tableName, "VALUES 10");
+            assertQuery("SELECT count(*) FROM \"" + tableName + "$partitions\"", "VALUES 8");
+            assertQuery("SELECT count(partition._varchar) FROM \"" + tableName + "$partitions\"", "VALUES 6");
+            assertQuery("SELECT count(distinct partition._date) FROM \"" + tableName + "$partitions\"", "VALUES 2");
+
+            assertQuery("SHOW COLUMNS FROM \"" + tableName + "$partitions\"",
+                    """
+                            VALUES
+                            ('partition', 'row(_varchar varchar, _date date)', '', ''),
+                            ('file_count', 'bigint', '', ''),
+                            ('total_size', 'bigint', '', ''),
+                            ('data', 'row(_bigint row(min bigint, max bigint, null_count bigint))', '', '')
+                            """);
+
+            assertQuery(
+                    "SELECT partition._varchar, partition._date FROM \"" + tableName + "$partitions\"",
+                    """
+                            VALUES
+                            ('b', DATE '2019-09-09'),
+                            ('c', DATE '2019-09-09'),
+                            ('d', DATE '2019-09-09'),
+                            ('e', DATE '2019-09-10'),
+                            ('f', DATE '2019-09-10'),
+                            (null, DATE '2019-09-10'),
+                            ('g', null),
+                            (null, null)
+                            """);
+
+            assertThat(query("SELECT CAST(data._bigint AS ROW(BIGINT, BIGINT, BIGINT)) FROM \"" + tableName + "$partitions\""))
+                    .matches("""
+                            VALUES 
+                            ROW(ROW(BIGINT '3', BIGINT '3', BIGINT '0')),
+                            ROW(ROW(BIGINT '1', BIGINT '1', BIGINT '0')),
+                            ROW(ROW(BIGINT '6', BIGINT '8', BIGINT '0')),
+                            ROW(ROW(BIGINT '7', BIGINT '7', BIGINT '0')),
+                            ROW(ROW(BIGINT '4', BIGINT '4', BIGINT '0')),
+                            ROW(ROW(BIGINT '6', BIGINT '6', BIGINT '0')),
+                            ROW(ROW(BIGINT '4', BIGINT '50', BIGINT '0')),
+                            ROW(ROW(BIGINT '2', BIGINT '2', BIGINT '0'))
+                            """);
+        }
+        finally {
+            assertUpdate("DROP TABLE IF EXISTS " + tableName);
+        }
+    }
+    
+    private void testPartitionsTableDataColumn(String type, String columnName, String insertIntoValues, int insertIntoValuesCount, String expectedDataColumn)
```

Comment:
Remove `columnName` and use static column name. 
Move this method under testPartitionsTableDataColumn.
Add `@Language("SQL")` annotation to insertIntoValues and expectedDataColumn.